ui/pages/notifications/notification-components/snap/snap.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import classnames from 'classnames';
+import { useSelector, useDispatch } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import type { SnapNotification } from '../../snap/types/types';
+import { SnapUIMarkdown } from '../../../../components/app/snaps/snap-ui-markdown';
+import { formatDate, getSnapRoute } from '../../../../helpers/utils/util';
+import { getSnapMetadata } from '../../../../selectors';
+import { markNotificationsAsRead } from '../../../../store/actions';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import Button from '../../../../components/ui/button';
+
+type SnapComponentProps = {
+  snapNotification: SnapNotification;
+};
+
+function NotificationItem({ snapNotification }: SnapComponentProps) {
```

Comment:
This is the original snap notification item. Mostly ported over from JS to TS (which found some bugs) - we kept as is (using `@ts-expect-error`). These fixes, alongside a minor redesign that both teams are satisfied with, will come in a separate PR.